src/font.cpp

```diff
@@ -8,6 +8,13 @@ Font::Font(const std::string& path, int size) {
     }
 }
 
+Font::Font(const std::filesystem::path& path, int size) {
+    m_font = TTF_OpenFont(path.string().c_str(), size);
+    if (m_font == nullptr) {
+        throw FontLoadingError{ "error loading font: '" + std::string{ TTF_GetError() } + "'" };
+    }
+}
+
```

Comment:
It should be possible to just call the other constructor instead.
```suggestion
Font::Font(const std::filesystem::path& path, int size) : Font{ path.string(), size } { }

```